gateway-server/src/main/java/org/apache/knox/gateway/GatewayServlet.java

```diff
@@ -277,4 +278,34 @@ public Enumeration<String> getInitParameterNames() {
       return config.getInitParameterNames();
     }
   }
+
+  private Exception sanitizeException(Exception e) {
+    if (e == null || e.getMessage() == null) {
+      return e;
+    }
+    if (!isErrorMessageSanitizationEnabled || e.getMessage() == null) {
+      return e;
+    }
+    String sanitizedMessage = e.getMessage().replaceAll("\\b(?:\\d{1,3}\\.){3}\\d{1,3}\\b", "[hidden]");
+    return createNewException(e, sanitizedMessage);
+  }
+
+  private <T extends Exception> T createNewException(T e, String sanitizedMessage) {
+    try {
+      Constructor<? extends Exception> constructor = e.getClass().getConstructor(String.class, Throwable.class);
+      T sanitizedException = (T) constructor.newInstance(sanitizedMessage, e.getCause());
+      sanitizedException.setStackTrace(e.getStackTrace());
+      return sanitizedException;
+    } catch (Exception ex) {
+      Exception genericException = new Exception(sanitizedMessage, e.getCause());
+      genericException.setStackTrace(e.getStackTrace());
+      return (T) genericException;
+    }
+  }
+
+  private <T extends Exception> T sanitizeAndRethrow(Exception e) throws T {
+    Exception sanitizedException = sanitizeException(e);
+    LOG.failedToExecuteFilter(sanitizedException);
+    throw (T) sanitizedException;
```

Comment:
Why does this have to throw rather than simply return the sanitized exception?

Reply:
@pzampino I chose this approach because, unlike returning an exception, which requires upstream code to handle the `Exception` type, throwing it directly takes advantage of Java's compiler to manage error handling more effectively, avoiding the need for callers to explicitly deal with broad `Exception` handling.

Reply:
It seems to me that all the callers of this method invoke it as `throw sanitizeAndRethrow(e) ` ; If this method returned the new exception object, it would be thrown by those invocations, no?

Reply:
@pzampino Nope! I was wrong. Not sure what happened. My guess is that there was an issue with my IDE.